Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.h

```diff
@@ -247,6 +247,33 @@ class WebsiteDataStoreConfiguration : public API::ObjectImpl<API::Object::Type::
     void setMemoryFootprintNotificationThresholds(Vector<size_t>&& thresholds) { m_memoryFootprintNotificationThresholds = WTFMove(thresholds); }
     const Vector<size_t>& memoryFootprintNotificationThresholds() { return m_memoryFootprintNotificationThresholds; }
 
+    struct Directories {
+        String applicationCacheFlatFileSubdirectoryName { "Files"_s };
+        String applicationCacheDirectory;
+        String alternativeServicesDirectory;
+        String cacheStorageDirectory;
+        String cookieStorageFile;
+        String deviceIdHashSaltsStorageDirectory;
+        String generalStorageDirectory;
+        String hstsStorageDirectory;
+        String indexedDBDatabaseDirectory;
+        String javaScriptConfigurationDirectory;
+        String localStorageDirectory;
+        String mediaCacheDirectory;
+        String mediaKeysStorageDirectory;
+        String networkCacheDirectory;
+        String resourceLoadStatisticsDirectory;
+        String searchFieldHistoryDirectory;
+        String serviceWorkerRegistrationDirectory;
+        String webSQLDatabaseDirectory;
+#if ENABLE(ARKIT_INLINE_PREVIEW)
+        String modelElementCacheDirectory;
+#endif
+        Directories isolatedCopy() const&;
```

Comment:
Given the optimization I suggested above, and given that this struct only contains Strings (which can avoid copying when passing cross thread in some cases when we have a r-value reference), could you please add an `&&` overload (like we have for String).